src/Storages/Kafka/KafkaBlockInputStream.cpp

```diff
@@ -100,7 +103,22 @@ Block KafkaBlockInputStream::readImpl()
             switch (status)
             {
                 case IProcessor::Status::Ready:
-                    input_format->work();
+                    try
+                    {
+                        input_format->work();
+                    }
+                    catch (const Exception & e)
```

Comment:
1) I think it will be clearer to catch exception higher (where read_kafka_message is called).
2) after exception you need to do `input_format->resetParser(); ` otherwise you keep the parser in dirty state.

Reply:
3. some rows may already be pushed before exception. For me, it doesn't sound sane, I would prefer to have atomicity here - either the whole message was parsed, the whole was failed.

Reply:
I'll send you some draft commit with proposal.

Reply:
Thank you.

Reply:
https://github.com/fastio/ClickHouse/pull/1

Reply:
That's very nice of you.